src/Aspire.Hosting.Dapr/DaprDistributedApplicationLifecycleHook.cs

```diff
@@ -267,25 +267,13 @@ public async Task BeforeStartAsync(DistributedApplicationModel appModel, Cancell
     {
         if (resource is IResourceWithEndpoints resourceWithEndpoints)
         {
-            if (sidecarOptions == null || (sidecarOptions.AppProtocol == null && sidecarOptions.AppEndpoint == null))
-            {
-                return (resourceWithEndpoints.GetEndpoint("http"), "http");
-            }
-            else if (sidecarOptions.AppProtocol == null && sidecarOptions.AppEndpoint != null)
-            {
-                var appEndpoint = resourceWithEndpoints.GetEndpoint(sidecarOptions.AppEndpoint);
-                return (appEndpoint, appEndpoint.Scheme);
-            }
-            else if (sidecarOptions.AppProtocol != null && sidecarOptions.AppEndpoint == null)
-            {
-                var appEndpoint = resourceWithEndpoints.GetEndpoint(sidecarOptions.AppProtocol);
-                return (appEndpoint, sidecarOptions.AppProtocol);
-            }
-            else if (sidecarOptions.AppProtocol != null && sidecarOptions.AppEndpoint != null)
-            {
-                var appEndpoint = resourceWithEndpoints.GetEndpoint(sidecarOptions.AppEndpoint);
-                return (appEndpoint, sidecarOptions.AppProtocol);
-            }
+            return (sidecarOptions, resourceWithEndpoints) switch{
```

Comment:
Why not get rid of teh when expressions and just check the 2 values? That makes it much cleaner.

Reply:
how does that work? It's my first time with the new switch expression :O

Reply:
It's the same as the old `switch` with less "boilerplate" if you like. You may find [these docs](https://learn.microsoft.com/dotnet/csharp/language-reference/operators/switch-expression) helpful.

In a nutshel, this method switches on a number of different cases (quite elaborate) where args are typles and for each case returns tuples.

Does this makes sense?

Reply:
Sadly no. I did read the documentation but have no clue how it should work with two a parameters

Reply:
can you provide an example, maybe with the code from this pr

Reply:
My comment here described it https://github.com/dotnet/aspire/pull/3184#discussion_r1560554150

Reply:
sorry didn't saw this comment. Fixed it now.